pkg/maps/configmap/cell.go

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package configmap
+
+import (
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/maps"
+)
+
+// Cell initializes and manages the config map.
+var Cell = cell.Module(
+	"config-map",
+	"eBPF map config contains runtime configuration state for the Cilium datapath",
+
+	cell.Provide(newMap),
+)
+
+func newMap(lifecycle hive.Lifecycle) MapOut {
+	configmap := newConfigMap()
+
+	lifecycle.Append(hive.Hook{
+		OnStart: func(startCtx hive.HookContext) error {
+			return configmap.init()
+		},
+		OnStop: func(stopCtx hive.HookContext) error {
+			return configmap.close()
+		},
+	})
+
+	return MapOut{
+		ConfigMap: configmap,
+		BpfMap:    configmap,
+	}
+}
+
+type MapOut struct {
```

Comment:
I usually try to avoid writing `cell.Out`s with group values for the fear of typoing the group name. You can get around this with a helper, e.g. `func NewMapOut[Map any](m Map) MapOut[Map]`, but perhaps not worth adding it just yet.